pkg/operator/operator.go

```diff
@@ -282,3 +287,21 @@ func WaitForCRDs(ctx context.Context, timeout time.Duration, config *rest.Config
 	log.Info("all required CRDs are available")
 	return nil
 }
+
+// ensureToken ensures we can get a token for the Azure environment. Note that this doesn't actually
+// use the token for anything, it just checks that we can get one.
+func ensureToken(cfg *auth.Config) error {
+	creds, err := azidentity.NewDefaultAzureCredential(nil)
```

Comment:
@comtalyst maybe we should have a unified usage of constructing credentials to use here. I believe in managed mode we should only use ManagedIdentityCredential . The worst part of ``NewDefaultAzureCredential` is it could block for CLI input when it fallbacks to AzCLI login (if the environment has shell and azcli installed)

Reply:
Yeah we have a backlog item to fix this and so I didn't take a crack at doing anything better here. Agree with what you're saying though that we should have our own `GetCredential` that makes a ChainedCredential w/ the options we support.

Reply:
Ok I ended up sorta needing to almost fix this to fix @bcho's other comment. There's now a single method in operator that gets the token. It still uses default credential but we could swap it to something else very easily in a follow up PR.